### DISCUSSION TITLE
How to generate 3D-PNM from 2D image in porespy?

### DISCUSSION THREAD

**Discussion by likeduck** [PMEAL/porespy#773] [Q&A] [answered]
Dear prof,

  I have tried `snow_advanced`, but it requires 3D image input. So it not reah my goal.
  
  I  also try the `snow2`(Since the doc is not released yet, I don 't know if the input is 2D image or 3D, I use 2D).
here is my problem:

![捕获](https://user-images.githubusercontent.com/117707506/209457126-d476ec2c-d739-43f6-90eb-82cee98a1d36.PNG)

**Comment by jgostick** [comment, marked as answer]
Snow2 works on 2d images just find.  The error your getting is because ``phases`` is ``None``.  Are you passing in all the arguments as "named" or "keyword" arguments?  If you put them in the wrong order, without names, then some of the defaults might be overriding your input.

Also, thanks for reminding me that the snow2 example is still listed as work in progress.  I'll add that to our todo list asap.

**Reply by likeduck** [reply]
Due to some reasons to delay the work process, I am sorry not timely reply, thank you for your guidance !
